packages/plugin/src/browser/hosted-plugin-manager-client.ts

```diff
@@ -0,0 +1,133 @@
+import { WindowService } from '@theia/core/lib/browser/window/window-service';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import { FileDialogFactory, DirNode } from '@theia/filesystem/lib/browser';
+import { HostedPluginServer } from '../common/plugin-protocol';
+import { HostedPluginCommands } from './plugin-api-frontend-contribution';
+
+/**
+ * Responsible for UI to set up and control Hosted Plugin Instance.
+ */
+@injectable()
+export class HostedPluginManagerClient {
+    @inject(HostedPluginServer)
+    protected readonly hostedPluginServer: HostedPluginServer;
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+    @inject(FileDialogFactory)
+    protected readonly fileDialogFactory: FileDialogFactory;
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+    @inject(WindowService)
+    protected readonly windowService: WindowService;
+    @inject(FileSystem)
+    protected readonly fileSystem: FileSystem;
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    protected pluginLocation: URI | undefined;
+    protected pluginInstanceUri: string | undefined;
+
+    async start(): Promise<void> {
+        if (!this.pluginLocation) {
+            await this.selectPluginPath();
+            if (!this.pluginLocation) {
+                // selection was cancelled
+                return;
+            }
+        }
+        try {
+            this.messageService.info('Starting hosted instance server ...');
+            await this.doRunRequest(this.pluginLocation);
+            this.messageService.info('Hosted instanse is running at: ' + this.pluginInstanceUri);
+        } catch (error) {
+            this.messageService.error('Failed to run hosted plugin instanse: ' + this.getErrorMessage(error));
+        }
+    }
+
+    async stop(): Promise<void> {
+        try {
+            await this.hostedPluginServer.terminateHostedPluginInstance();
+            this.messageService.info((this.pluginInstanceUri ? this.pluginInstanceUri : 'The instance') + ' has been terminated.');
+        } catch (error) {
+            this.messageService.warn(this.getErrorMessage(error));
+        }
+    }
+
+    async restart(): Promise<void> {
+        if (await this.hostedPluginServer.isHostedTheiaRunning()) {
+            await this.stop();
+
+            this.messageService.info('Starting hosted instance server ...');
+            // It takes some time before OS released all resources e.g. port.
+            // Keeping tries to run hosted instance with delay.
+            let lastError;
+            for (let tries = 0; tries < 15; tries++) {
+                try {
+                    await this.doRunRequest(this.pluginLocation!);
+                    this.messageService.info('Hosted instanse is running at: ' + this.pluginInstanceUri);
```

Comment:
there is a typo on `instance`

Reply:
Yeap, again) I have to stop copy paste old messages. Thanks.